core/templates/dev/head/pages/exploration_editor/statistics_tab/EarlyQuitIssueDirective.js

```diff
@@ -0,0 +1,157 @@
+        index: '&',
+        // A read-only object representing the issue.
+        issue: '&'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_editor/statistics_tab/' +
+        'early_quit_issue_directive.html'),
+      controller: [
+        '$scope', '$uibModal', 'IssuesService',
+        function($scope, $uibModal, IssuesService) {
+          $scope.currentIssueIdentifier = $scope.index() + 1;
+
+          var issue = $scope.issue();
+          $scope.issueStatement = IssuesService.renderIssueStatement(issue);
+          $scope.suggestions = IssuesService.renderIssueSuggestions(issue);
+          $scope.playthroughIds = issue.playthroughIds;
+
+          var getPlaythroughIndex = function(playthroughId) {
+            return $scope.playthroughIds.indexOf(playthroughId);
+          };
+
+          $scope.createPlaythroughNavId = function(playthroughId) {
+            return getPlaythroughIndex(playthroughId) + 1;
+          };
+
+          $scope.showPlaythrough = function(playthroughId) {
+            IssuesService.getPlaythrough(
+              playthroughId
+            ).then(function(playthrough) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/pages/exploration_editor/statistics_tab/' +
+                  'playthrough_modal_directive.html'),
+                backdrop: true,
+                resolve: {
+                  playthrough: function() {
+                    return playthrough;
+                  },
+                  playthroughIndex: function() {
+                    return $scope.playthroughIds.indexOf(playthroughId);
+                  }
+                },
+                controller: [
+                  '$scope', '$uibModalInstance', 'playthroughIndex',
+                  'playthrough', 'AlertsService', 'LearnerActionRenderService',
+                  function(
+                      $scope, $uibModalInstance, playthroughIndex,
+                      playthrough, AlertsService, LearnerActionRenderService) {
+                    $scope.playthroughIndex = playthroughIndex;
+
+                    $scope.displayBlocks =
+                      LearnerActionRenderService.getDisplayBlocks(
+                        playthrough.actions);
+
+                    var blockActionIndexMapping = {};
+                    var i = 0;
+                    var total = $scope.displayBlocks.reduce(
+                      function(runningTotal, displayBlock) {
+                        blockActionIndexMapping[i] = runningTotal;
+                        i += 1;
+                        return runningTotal + displayBlock.length;
+                      }, 1);
+
+                    $scope.maxHidden = $scope.displayBlocks.length - 1;
+
+                    $scope.getDisplayBlockIndex = function(displayBlock) {
+                      return $scope.displayBlocks.indexOf(displayBlock);
```

Comment:
I'm still a bit concerned about doing so much work just for an index. Does the UI lag at all?

Reply:
SGTM. I'm not convinced we can't do this using index, but I'm fine with leaving that refactoring as a TODO.